include/ck/utility/inner_product.hpp

```diff
@@ -84,6 +84,18 @@ __device__ void inner_product<half_t, half_t, float>(const half_t& a, const half
     inner_product(type_convert<float>(a), type_convert<float>(b), c);
 }
 
+template <>
+__device__ void inner_product<bhalf2_t, bhalf2_t, float>(const bhalf2_t& a, const bhalf2_t& b, float& c)
+{
+    const vector_type<bhalf_t, 2> a_vector{a};
+    const vector_type<bhalf_t, 2> b_vector{b};
+
+    static_for<0, 2, 1>{}([&](auto i) {
+        c += type_convert<float>(a_vector.AsType<bhalf_t>()[i]) *
```

Comment:
Please use inner_product<float, float>, it allows to use builtin or assembly instruction when enabled

Reply:
Will do, thanks!

Reply:
Actually, the bhalf case already does that, so we can delegate bhalf2 to bhalf. Hope that works